### PR TITLE
HADOOP-16826. ABFS: update abfs.md to include config keys for identity transformation

### PR DESCRIPTION
**Pull request by karthick-rn**
In this PR, I have updated the `abfs.md` to include some key configurations required when submitting jobs to YARN using hadoop-azure module. Currently, these properties aren't documented anywhere in Apache Hadoop. I have them included under 'Troubleshooting' section of `abfs.md`, please review and also let me know if you think they need to be documented elsewhere?  
As the PR is only an update to the existing doc I have not opened any JIRA ticket, let me know if you think a JIRA is still required? 

Many thanks!

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 23s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  20m 38s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 30s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  34m 50s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 25s |  the patch passed  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 2 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  shadedclient  |  15m  3s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 28s |  The patch does not generate ASF License warnings.  |
|  |   |  54m  7s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1785/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1785 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux 588f6773d25a 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 926d0b4 |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-1785/1/artifact/out/whitespace-eol.txt |
| Max. process+thread count | 329 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1785/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  25m  9s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  18m 17s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 35s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  31m 28s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 26s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  13m 23s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   |  72m 57s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1785/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1785 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux 666ceee34167 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 926d0b4 |
| Max. process+thread count | 412 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1785/2/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-tools/hadoop-azure/src/site/markdown/abfs.md [hunk 1]
```diff
@@ -857,6 +857,48 @@ signon page for humans, even though it is a machine calling.
 1. The URL is wrong —it is pointing at a web page unrelated to OAuth2.0
 1. There's a proxy server in the way trying to return helpful instructions.
 
+### `java.io.IOException: The ownership on the staging directory /tmp/hadoop-yarn/staging/user1/.staging is not as expected. It is owned by <principal_id>. The directory must be owned by the submitter user1 or user1`
+
+When using [Azure Managed Identities](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview), the files/directories in ADLS Gen2 are by default owned by the service principal object id i.e. principal ID & submitting jobs as the local OS user 'user1' results in the above exception.
+
+The following configurations have to be added to core-site.xml to resolve this issue
+
+```
```
Comment (steveloughran) [inline hadoop-tools/hadoop-azure/src/site/markdown/abfs.md]: declare as xml for some tools to render better

**Comment by steveloughran** [comment]
LGTM; one minor tweak and then it's good to go in. Troubleshooting docs are, provided kept current, wonderful

File: hadoop-tools/hadoop-azure/src/site/markdown/abfs.md (hunk 1 above)
Comment (karthick-rn) [reply hadoop-tools/hadoop-azure/src/site/markdown/abfs.md]: done. Thanks @steveloughran

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  25m 40s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  18m 11s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 34s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  31m 22s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 27s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 39s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  13m 28s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 32s |  The patch does not generate ASF License warnings.  |
|  |   |  73m 36s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1785/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1785 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux 96d349367a81 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 263413e |
| Max. process+thread count | 412 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1785/3/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-tools/hadoop-azure/src/site/markdown/abfs.md [hunk 2]
```diff
@@ -857,6 +857,46 @@ signon page for humans, even though it is a machine calling.
...
+  <description>
+  A comma separated list of names to be replaced with the service principal ID specified by
+  “fs.azure.identity.transformer.service.principal.id”.  This substitution occurs
+  when setOwner, setAcl, modifyAclEntries, or removeAclEntries are invoked with identities
+  contained in the substitution list. Notice that when in non-secure cluster, asterisk symbol *
+  can be used to match all user/group.
+  </description>
+</property>
+<property>
+  <name>fs.azure.use.upn</name>
```
[the remark this answers is not in the document]
Comment (karthick-rn) [reply hadoop-tools/hadoop-azure/src/site/markdown/abfs.md]: @snvijaya Removed this config in the new commit

File: hadoop-tools/hadoop-azure/src/site/markdown/abfs.md [hunk 3]
```diff
@@ -857,6 +857,46 @@ signon page for humans, even though it is a machine calling.
 1. The URL is wrong —it is pointing at a web page unrelated to OAuth2.0
 1. There's a proxy server in the way trying to return helpful instructions.
 
+### `java.io.IOException: The ownership on the staging directory /tmp/hadoop-yarn/staging/user1/.staging is not as expected. It is owned by <principal_id>. The directory must be owned by the submitter user1 or user1`
+
```
[the remark this answers is not in the document]
Comment (karthick-rn) [reply hadoop-tools/hadoop-azure/src/site/markdown/abfs.md]: @snvijaya The options suggested are not a suitable fix for the exception.  I think `identity.transformer` is more of a fix rather to be considered as a workaround. I have added a short message on the fix in my last commit. Let me know for any further questions?

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  4s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  26m 13s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 34s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  43m 10s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 29s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 29s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  17m 38s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 35s |  The patch does not generate ASF License warnings.  |
|  |   |  66m 30s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1785/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1785 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux bc4ee17148c8 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / f206b73 |
| Max. process+thread count | 306 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1785/4/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by karthick-rn** [comment]
@steveloughran @snvijaya - Please can you review the changes and merge if you're happy? Thanks!

**Comment by DadanielZ** [comment]
committed, thank you @karthick-rn

**Pull request closed without merging**